### ISSUE TITLE
[2.2] Creation of Software Design

### ISSUE THREAD

**Issue by ImUrGit** [lowyisan/INF2001_P6-6#83]
What is the task about? This task is focused on documenting supplementary system designs which include the Overall Class Diagram, Software Component Diagram, and Sequence Diagram.

What are the goals of the task? Ensure that all relevant supplementary system designs are documented comprehensively.

Success or Completion Criteria: Provide well-documented and up-to-date supplementary system designs, addressing the specific areas of concern.

Allocated time: 9 days

Start date and end date: [Start Date: 20/10/23], [End Date: 28/10/23]

Owner: All

Status of the task: To-Do

Priority: Medium

**Comment by mvchello** [comment]
**Describe the work done in detail.**
Aided in the creation of the Component diagram. Helped to ensure the components aligned with the other diagrams that were created.

**Has this work been committed to the repository?**
No work is needed to commit to the repository.

**How does the work affect success or completion criteria?**
It is important to make such diagrams as it helps the team and clients to be on the same page regarding the structure of the product.

**Start date and end date:**
[Start Date: 20/10/23]
[End Date: 28/10/23]

**Comment by Magic1sarap** [comment]
**Describe the work done in detail.**
Created the sequence diagram for UC15 (Use Case 15: Manage new job assigned).

**Has this work been committed to the repository?**
No, as no code was involved. 

**How does the work affect success or completion criteria?**
The creation of the sequence diagram will help the viewers of the document to understand the system flow for when staff accept or reject jobs as well as recommending a different staff after job rejection. 

**Start date and end date:**
[Start Date: 20/10/23]
[End Date: 28/10/23]

**Comment by ImUrGit** [comment]
**Describe the work done in detail.**
I have helped in creating the component diagram together with Michelle. I have also ensured that the components reflect the classes in the created class diagram.

**Has this work been committed to the repository?**
No work is needed to commit to the repository.

**How does the work affect success or completion criteria?**
It is important to have a component diagram that is reflective of our system as it will help our stakeholders get an idea of what the system is capable of.

**Start date and end date:**
[Start Date: 20/10/23]
[End Date: 28/10/23]

**Comment by ThJin** [comment]
**Describe the work done in detail.**
The work involved the creation of the overall class diagram, ensuring that they are well-detailed.

**Has this work been committed to the repository?**
The work does not involve any code. Therefore, no repository commits is needed.

**How does the work affect success or completion criteria?**
The work provides a clear and comprehensive design on the system's architecture and functionality.

Start date and end date:
[Start Date: 20/10/23]
[End Date: 28/10/23]

**Comment by KeeHanXiang** [comment]
Describe the work done in detail.
The work involved the creation of the sequence diagram.

Has this work been committed to the repository?
The work does not involve any code. Therefore, no repository commits is needed.

How does the work affect success or completion criteria?
The work provides a clear understanding of the user's actions and method calls in the system.

Start date and end date:
[Start Date: 20/10/23]
[End Date: 28/10/23]

**Comment by lowyisan** [comment]
Describe the work done in detail:
For this task, I was responsible for documenting the Overall Class Diagram. I conducted a comprehensive review of our system's design, ensuring that all relevant classes, attributes, methods, and relationships were accurately represented. I also made updates based on the feedback from our M1 report.

Has this work been committed to the repository?
The work does not involve any code. Therefore, no repository commits is needed.

How does the work affect success or completion criteria?
The Overall Class Diagram is a crucial component of our software design, providing a clear representation of the system's structure. By revising and updating it, we ensure that our system design aligns with the feedback received during the M1 review. This contributes to the overall success and completion of our Milestone 2 goals.

Start date and end date:
[Start Date: 20/10/23]
[End Date: 28/10/23]
